patterns/immediate-nullification/src/test/e2e.test.ts

```diff
@@ -0,0 +1,136 @@
+import {
+    AccountWalletWithPrivateKey,
+    createPXEClient,
+    ExtendedNote,
+    Fr,
+    PXE,
+    waitForPXE,
+  } from "@aztec/aztec.js";
+  
+  import { ImmediateNoteNullificationContract } from "../../../../artifacts/immediate-nullification/ImmediateNoteNullification.js";
+  import { getInitialTestAccountsWallets } from "@aztec/accounts/testing";
+  
+  // Global variables
+  let pxe: PXE;
+  let immediateNullificationContract: ImmediateNoteNullificationContract;
+  
+  let user: AccountWalletWithPrivateKey;
+  let deployer: AccountWalletWithPrivateKey;
+  
+  const { PXE_URL = 'http://localhost:8080' } = process.env;
+  
+  const setupSandbox = async () => {
+    const pxe = createPXEClient(PXE_URL);
+    await waitForPXE(pxe);
+    return pxe;
+  };
+  
+  // Setup: Set the sandbox
+  beforeAll(async () => {
+    pxe = await setupSandbox();
+    [user, deployer] = await getInitialTestAccountsWallets(pxe);
+
+    }, 120_000);
+
+  
+  describe("E2E Example Note", () => {
+    let randomness: Fr;
+    beforeAll(async () => {
+      const deploymentReceipt = await ImmediateNoteNullificationContract.deploy(deployer)
+        .send()
+        .wait();
+  
+      immediateNullificationContract = deploymentReceipt.contract;
+    }, 200_000);
+
+    describe("create_note(...)", () => {
+        let exampleNotes: ExtendedNote[];
+
+        it("should mine the transaction", async () => {
```

Comment:
"transaction should not revert" or something similar? iirc, they left a comment once about it - what we care is not reverting (i know, I nit now)